ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/vm/OperationRegistry.java

```diff
@@ -12,31 +12,39 @@
  */
 package tech.pegasys.pantheon.ethereum.vm;
 
+import com.google.common.base.Preconditions;
+
 /** Encapsulates a group of {@link Operation}s used together. */
 public class OperationRegistry {
 
   private static final int NUM_OPERATIONS = 256;
 
-  private final Operation[] operations;
+  private final Operation[][] operations;
 
   public OperationRegistry() {
-    this.operations = new Operation[NUM_OPERATIONS];
+    this(1);
+  }
+
+  public OperationRegistry(final int numVersions) {
+    Preconditions.checkArgument(numVersions >= 1);
+    this.operations = new Operation[numVersions][NUM_OPERATIONS];
```

Comment:
(Optional) You might look into guava's `RangeMap`, which could map a range of versions to a single operation, rather than creating distinct references for every op-version combo.

Reply:
Since this is a "hotspot" I opted for native arrays, on the presumption native arrays are faster than interfaces.  When we get better performance monitoring in we can see how much the impact is.